### DISCUSSION TITLE
Provide a transform option on `useTina` for active field indicator`

### DISCUSSION THREAD

**Discussion by jeffsee55** [tinacms/tinacms#2609] [Ideas]
The ergonomics of adding the `data-tinaField` property can be pretty awkward once your objects start to get more complex. I'm wondering if we should provide a property on `useTina` that would transform the payload in to an easier format for attaching these elements (note: `$id` and `$value` can be whatever we decide, just making them clear for demonstration):

```jsx
// terrible name, but the idea is setting some flag would return a slightly different
// shape for the `data` property
const { data } = useTina(query, variables, data, {activeFieldProperties: true})

return (
  <div>
    
    <h2 data-tinaField={data.title.$id}>
      {data.title.$value}
    </h2>
    <ul data-tinaField={data.features.$id}>
      {data.features.$value.map(feature => (
        <li data-tinaField={feature.$id}>{feature.$value}</li> 
      ))}
    </ul>
  </div>
)
```

Obviously our generated types would need to take this into consideration, too - but I think that's doable.

**Comment by jamespohalloran** [comment]
~~Seems pretty smooth! Could/Should activeFieldProperties be a part of the query instead of a useTina param?~~
Edit: Nope

**Reply by jeffsee55** [reply]
@jamespohalloran I'm not sure what you mean?

**Reply by jamespohalloran** [reply]
Just wondering why we'd need this option: `{activeFieldProperties: true}`. Would `$id` just be apart of the regular query/response? Or is this something are generating within useTina client-side?

**Reply by jeffsee55** [reply]
I was suggesting that it wouldn't alter the query/response in any way, just a transform that happens client-side to keep things simple. So - no, $id and $value wouldn't come from the server at all, do you see an advantage to that? It seems like that might complicate the GraphQL API in unnecessary ways

**Reply by jamespohalloran** [reply]
Right right, yeah makes sense!

**Comment by jeffsee55** [comment]
[Example type implementation](https://www.typescriptlang.org/play?#code/C4TwDgpgBACg9gZ2FAvFA3gWAFBSsAS2ABsIAuKJAJwIDsBzAGhzwCM4ATEC6u+lqAEMArsAAWcKhQCCoiVWa4oYYa2IEEYiBwrs4pQbRwBfHDlCQos8ZNQYBtQQFtylYDQaK8E1qwIQEHnc+AG0AXRMzbHNwaAB1IjEASQ4AHgAVAD47LCUQgGkoOigAawgQOAAzKHSwinQAEgA3QWJhVwTxFIyCsMzGKAaCHTcPemMAbhMp6OwAejmoAGM4WiR8KkMESsknOwyoCAAPYAhaDgQoOFYAKwgl4EyACmubinSASgpO5LSs1GyuQWeCgVAgwGEVFoV1uOGBplmKzWyHcWx2VD2aAOx1O50ur3ujxet3eXwwQxGvE8gxabQ6iW6WWMAPsSmBAAFgAgALQEei0SQQAbiDRFBC0ADkyEED2ErWIICgAHdJCU+AMbsJ1uj8LEoBwIE5VtRBIRVgIwRCoTCbpFEcbkGBEMAKPB1mhcnhCCRXAAiLTEOC+rxQdhcCgSlVUYgcCUhlRqDRaEbudohkQ2KSskFQRwuCN3SqVOMCbzXPwBMghCVwEoSiJKBEInBI9YuMR2VFrdFOJ5OpAfHDtgB0zVa7WHGfko9pE58FYQM-HECAA) - doesn't actually work but you can see how the types would work
